__test__/api/applications/deleteSingleApplication.test.js

```diff
@@ -0,0 +1,50 @@
+const app = require("../../../server");
+const supertest = require("supertest");
+const mongoose = require("mongoose");
+const ApplicationModel = require("../../../models/applications");
+const request = supertest(app);
+
+describe("Delete /applications/:applicationId", () => {
+  let application;
+  beforeEach(async () => {
+    application = new ApplicationModel({
+      name: "my app",
+      organizationId: global.organization._id,
+      createdBy: global.admin._id,
+    });
+    await application.save();
+  });
+
+  afterEach(async () => {
+    await ApplicationModel.findByIdAndDelete(application._id);
+    application = null;
+  });
+
+  it("Deletes application with provided ID", async () => {
+    let res = await request
+      .delete(`/v1/applications/${application._id}`)
+      .set("Authorization", `bearer ${global.orgToken}`);
+    expect(res.status).toEqual(200);
+    expect(res.body.data._id).toEqual(String(application._id));
```

Comment:
According to the swagger documentation, the returned response should be using `applicationId` as the name for the id.
```suggestion
    expect(res.body.data.applicationId).toEqual(String(application._id));
```

Reply:
i will have to edit the controller then